### PR TITLE
New required transcript attrib for all canonical transcript

### PR DESCRIPTION
**Pull request by marcoooo**
Added DC for is_canonical transcript attrib.

### CODE REVIEW

**Review by luca-drf — APPROVED** [review]

**Comment by marcoooo** [comment]
Tested on a 106 database after patch:
```
perl run_datachecks.pl -dbname homo_sapiens_core_106_38 -registry_file ~/registries/st1.pm -name TranscriptSupport $(st1 details script) 

# Subtest: TranscriptSupport
    # Subtest: homo_sapiens, core, homo_sapiens_core_106_38
        ok 1 - All genes have at least one transcript with a gencode_basic attribute
        ok 2 - All genes have at least one transcript with a is_canonical attribute
        ok 3 - TSL attributes exist
        ok 4 - CCDS xrefs exist
        1..4
    ok 1 - homo_sapiens, core, homo_sapiens_core_106_38
    1..1
ok 1 - TranscriptSupport
1..1
ok
All tests successful.
Files=1, Tests=1,  3 wallclock secs ( 0.04 usr +  0.01 sys =  0.05 CPU)
Result: PASS
```

**Review by gnaamati — COMMENTED** [review]
Sorry for late reply, can you explain briefly what the change is?
Guy

**Comment by jmgonzmart** [comment]
It looks good to me.

File: lib/Bio/EnsEMBL/DataCheck/Checks/TranscriptSupport.pm [hunk 1]
```diff
@@ -48,24 +48,53 @@ sub tests {
...
+  my $attribs_count = $helper->execute( -SQL => $sql_1b );
+  my $desc_1b = "is_canonical, genecode_basic exists in transcript_attrib set";
+  my $detail_desc;
+  # Expect exactly two lines returned even if zero lines, one per attrib.code
+  cmp_ok(scalar @$attribs_count, '=', 2, $desc_1b);
+  is_rows_zero($self->dba, $sql_1c, $desc_1c);
+  foreach my $attrib_count (@$attribs_count) {
+    my ($count, $code) = @$attrib_count;
+    $detail_desc = sprintf($desc_1, $code);
+    ($count, $gene_count, $detail_desc);
```
Comment (luca-drf) [inline lib/Bio/EnsEMBL/DataCheck/Checks/TranscriptSupport.pm:95]: Is this a typo? Perhaps you meant:
```suggestion
    is($count, $gene_count, $detail_desc);
```

**Pull request merged**